core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java

```diff
@@ -37,29 +40,43 @@ public void testThreadAffinity(Class<?>... classes) {
     XmlSuite xmlsuite = createXmlSuite("test_suite");
     xmlsuite.setParallel(XmlSuite.ParallelMode.CLASSES);
     xmlsuite.setThreadCount(6);
-    createXmlTest(xmlsuite, "Test 1", classes);
+    createXmlTest(xmlsuite, "Test_1", classes);
     TestNG testng = create(xmlsuite);
     testng.run();
-    assertThat(LogGatheringListener.getLog()).hasSize(2);
+    Map<Class<?>, Set<Long>> map = LogGatheringListener.getLog().get("Test_1");
+    Arrays.stream(classes)
+        .forEach(each -> assertThat(map.get(each))
```

Comment:
Maybe you can use SoftAssert for that: https://assertj.github.io/doc/#assertj-core-soft-assertions

Reply:
fixed and pushed amended commit